src/test/java/com/google/jenkins/plugins/computeengine/integration/ITUtil.java

```diff
@@ -0,0 +1,238 @@
+  private static final String LAUNCH_TIMEOUT_SECONDS_STR = "";
+
+  static String projectId = format("%s");
+
+  private static String format(String s) {
+    String projectId = System.getenv("GOOGLE_PROJECT_ID");
+    if (projectId == null) {
+      throw new RuntimeException("GOOGLE_PROJECT_ID env var must be set");
+    }
+    return String.format(s, projectId);
+  }
+
+  static ComputeClient init(JenkinsRule r, StreamHandler sh, Map<String, String> label, Logger log)
+      throws Exception {
+    log.info("init");
+
+    assertNotNull("GOOGLE_PROJECT_ID env var must be set", projectId);
+
+    String serviceAccountKeyJson = System.getenv("GOOGLE_CREDENTIALS");
+    assertNotNull("GOOGLE_CREDENTIALS env var must be set", serviceAccountKeyJson);
+
+    ServiceAccountConfig sac = new StringJsonServiceAccountConfig(serviceAccountKeyJson);
+    Credentials c = new GoogleRobotPrivateKeyCredentials(projectId, sac, null);
+
+    CredentialsStore store = new SystemCredentialsProvider.ProviderImpl().getStore(r.jenkins);
+    assertNotNull("Credentials store can not be null", store);
+    store.addCredentials(Domain.global(), c);
+
+    // Add Cloud plugin
+    ComputeEngineCloud gcp = new ComputeEngineCloud(CLOUD_NAME, projectId, projectId, "10", null);
+
+    // Capture log output to make sense of most failures
+    Logger cloudLogger =
+        LogManager.getLogManager()
+            .getLogger("com.google.jenkins.plugins.computeengine.ComputeEngineCloud");
+    if (cloudLogger != null) {
+      cloudLogger.addHandler(sh);
+    }
+
+    assertEquals(0, r.jenkins.clouds.size());
+    r.jenkins.clouds.add(gcp);
+    assertEquals(1, r.jenkins.clouds.size());
+
+    // Get a compute client for out-of-band calls to GCE
+    ClientFactory clientFactory = new ClientFactory(r.jenkins, new ArrayList<>(), projectId);
+    ComputeClient client = clientFactory.compute();
+    assertNotNull("ComputeClient can not be null", client);
+
+    // Other logging
+    Logger clientLogger =
+        LogManager.getLogManager()
+            .getLogger("com.google.jenkins.plugins.computeengine.ComputeClient");
+    if (clientLogger != null) {
+      clientLogger.addHandler(sh);
+    }
+
+    deleteIntegrationInstances(true, client, label, log);
+    return client;
+  }
+
+  static void teardown(
+      StreamHandler sh,
+      ByteArrayOutputStream logOutput,
+      ComputeClient client,
+      Map<String, String> label,
+      Logger log)
+      throws IOException {
+    log.info("teardown");
+    deleteIntegrationInstances(false, client, label, log);
+    sh.close();
+    log.info(logOutput.toString());
+  }
+
+  static InstanceConfiguration instanceConfiguration(
+      String startupScript,
+      String numExecutors,
+      String jenkinsLabels,
+      Map<String, String> label,
+      boolean createSnapshot,
+      boolean oneShot,
+      String template) {
+    InstanceConfiguration ic =
+        new InstanceConfiguration(
```

Comment:
Tech debt note: the constructor for InstanceConfiguration has gotten large enough to warrant use of a builder.